### DISCUSSION TITLE
Enable usage of only one EDC for both Consumer and Provider Roles

### DISCUSSION THREAD

**Discussion by matbmoser** [eclipse-tractusx/tractusx-edc#869] [Ideas]
# Introduction
Hi I am @matbmoser commiter in the Tractus-X Project and Developer of the eclipse-tractusx/digital-product-pass application. In our use case we use the EDC as Consumer (https://github.com/eclipse-tractusx/digital-product-pass/tree/main/deployment/helm/edc-consumer) but for our test cases we actually need to test using a MOCK EDC Provider setup which is configured here: https://github.com/eclipse-tractusx/digital-product-pass/tree/main/deployment/helm/edc-provider.

I know that the EDC has currently >`v0.5.0` the differentiation between Consumer and Provider logs so I had the idea of using one EDC for both proposes Consumer and Provider.

However here comes the problem we tried to test one single EDC as "Consumer" and "Provider" however we were not able to go forward with that, since the tables in the Postgres DB where conflicting when one EDC had two roles.

I have not the specific error at the moment however just want to make a proposal: 

# Proposal
- Enable the usage of one EDC with the two roles being able to co-exist.

**Comment by ndr-brt** [comment]
A connector can actually be (and in the Tractus-X EDC, it is) "consumer" and "provider" in the same runtime, but not in the same negotiation/transfer flow, it wouldn't make any sense, and I bet such a proposal would be discarded by the EDC project.

**Comment by lgblaumeiser** [comment]
Closed as stale, please renew item if this is still an issue
